SETUP.md

```diff
@@ -0,0 +1,121 @@
+# Setup Guide
+
+This document describes how to setup all the dependencies to run the notebooks
+in this repository.
+
+Many computer visions scenarios are extremely computationlly heavy. Training a
+model often requires a machine that has a GPU, and would otherwise be too slow.
+We recommend using the GPU-enabled [Azure Data Science Virtual Machine (DSVM)](https://azure.microsoft.com/en-us/services/virtual-machines/data-science-virtual-machines/) since it comes prepared with a lot of the prerequisites needed to efficiently do computer vision.
+
+To scale up or to operationalize your models, we recommend setting up [Azure
+ML](https://docs.microsoft.com/en-us/azure/machine-learning/). Our notebooks
+provide instructions on how to use it.
+
+
+## Table of Contents
+
+1. [Compute Environment](#compute-environments)
+1. [System Requirements](#system-requirements)
+1. [Installation](#installation)
+1. [Tunneling](#tunneling)
+
+## Compute Environments
+
+Most computer vision scenarios require a GPU, especially if you're training a
+custom model. We recommend using a virtual machine to run the notebooks on.
+Specifically, we'll want one with a powerful GPU. The NVIDIA's Tesla V100 is a
+good choice that can be found in most Azure regions.
+
+The easiest way to get started is to use the [Azure Data Science Virtual Machine (DSVM)](https://azure.microsoft.com/en-us/services/virtual-machines/data-science-virtual-machines/). This VM will come installed with all the system requirements that are needed to run the notebooks in this repository. If you choose this option, you can skip the [System Requirements](#system-requirements) step in this guide as those requirements come pre-installed on the DSVM.
+
+Here are some ways you can create the DSVM:
+
+__Virtual Machine Builder__
+
+One easy way to create your DSVM is to use the [VM Builder](../contrib/vm_builder) tool located inside of the 'contrib' folder in the root directory of the repo. Simply run `python contrib/vm_builder/vm_builder.py` at the root level of the repo and this tool will preconfigure your virtual machine with the appropriate settings for working with this repository.
+
+__Provision a Data Science VM with the Azure Portal or CLI__
+
+You can also spin up a Data Science VM directly using the Azure portal. To do so, follow
+[this](https://docs.microsoft.com/en-us/azure/machine-learning/data-science-virtual-machine/dsvm-ubuntu-intro)
+link that shows you how to provision your Data Science VM through the portal.
+
+You can alternatively use the Azure command line (CLI) as well. Follow
+[this](https://docs.microsoft.com/en-us/cli/azure/azure-cli-vm-tutorial?view=azure-cli-latest)
+link to learn more about the Azure CLI and how it can be used to provision
+resources.
+
+## System Requirement
+
+__Requirements__
+
+* A machine running Linux >= 16.04 LTS or Windows
```

Comment:
Need to point out here (and maybe also at other places) that especially for image classification we recommend Linux. That's because PyTorch runs much slower on Windows, for image classification, training on Linux can be up to 10x faster. For object detection, the difference is much smaller, and around 20-30%. See our OneNote page for more detailed info.

Reply:
that is already noted a few paragraphs below